### PR TITLE
/reconnect changes

### PR DESCRIPTION
**Pull request by mjkzy**
/reconnect will not work unless you have at least joined one server. this should fix some issues/crashes

### CODE REVIEW

File: src/client/component/party.cpp [hunk 1]
```diff
@@ -25,6 +25,8 @@ namespace party
 		} connect_state;
 
 		std::string sv_motd;
+		
+		int hostDefined;
```
Comment (momo5502) [inline src/client/component/party.cpp:29]: Can make this a bool and move it into the connect_state

**Review by momo5502 — CHANGES_REQUESTED** [review]

File: src/client/component/party.cpp (hunk 1 above)
Comment (mjkzy) [reply src/client/component/party.cpp:29]: no clue why i didn't do a bool and i didn't think of moving it into connect_state. will do that now

**Comment by mjkzy** [comment]
should be good now ^

File: src/client/component/party.cpp [hunk 2]
```diff
@@ -277,7 +276,7 @@ namespace party
 				return;
 			}
 			
-			hostDefined = 0;
+			connect_state.hostDefined = false;
```
Comment (momo5502) [inline src/client/component/party.cpp:279]: don't do that

File: src/client/component/party.cpp [hunk 3]
```diff
@@ -22,11 +22,10 @@ namespace party
 		{
 			game::netadr_s host{};
 			std::string challenge{};
+			bool hostDefined;
```
Comment (momo5502) [inline src/client/component/party.cpp:25]: Just define this to false:
bool hostDefined{false};

File: src/client/component/party.cpp (hunk 2 above)
Comment (momo5502) [reply src/client/component/party.cpp:279]: do the thing i showed above

**Review by momo5502 — CHANGES_REQUESTED** [review]

**Review by momo5502 — APPROVED** [review]

**Pull request merged**